rfc/rfc-36/rfc-36.md

```diff
@@ -0,0 +1,605 @@
+| 1    | 0      | 0       | 1       | file2.log | 1024    | 0          | '2022-01-01 00:01:00' | '2022-01-01 00:01:00' |
+
+### Consistency
+
+**Pre:** 'version' of a hudi table is auto-increasing and updated with CAS. Each version value corresponds to a completed action.
+
+**About commit**
+
+1. Write client completes a commit by sending instant's state changing message to server. Only when server responds to the success msg, can client consider it is a successfully write.
+
+2. Server completes a commit by
+    1. save instant metadata like `HoodieCommitMetadata`
+    2. change the state to 'completed'
+    3. increase the table version and put a <version, instant_id> pair to the `tbl_version` table in the storage
+
+**About snapshot**
+
+1. The 'snapshot_status' in table `tbl_version` marks whether the commit has completed the snapshot synchronization. When the touching files of the commit are stored to `files` table, 'snapshot_status' can be 1 referred to sync successfully.
+
+2. Before reading / writing the hudi table, client will get partitions from server. The server will respond to client until all completed commits of the table finish their snapshot synchronization to keep consistency.
+
+### Extension
+
+#### 1. Partition pruning
+
+To an SQL engine, the optimizer does partition pruning to skip reading data from unneeded partitions. Accorrding to the WHERE clauses in sql, unneeded partitions can be eliminated. The following is how the hudi metastore server does this.
+
+Read a hudi table by sql like
+
+```SQL
+select count(*) from test_db.test_table where date='20220101' and hour = '00' and ts = '0'
+```
+
+and we can get the filter `where date='20220101' and hour = '00' and ts = '0'` from the sql engine when optimizing the sql.
+
+After the partition service receiving the `filter sql`
+
+1. build an expression tree, analyze the sql and get the partition column related conditions
+
+```SQL
+where date='20220101' and hour = '00'
+```
+
+2. generate a new sql filter
+
+```SQL
+where part_key_name='date' and part_key_val='20220101'
+where part_key_name='hour' and part_key_val='00'
+```
+
+3. query the partition related table with the new sql filter to get the eligible partitions
+
+#### 2.Hive Metastore Adapter
+
+The adapter in hudi metatore will implement some commonly used API in hive metastore server to do the compatibility so that there are no changes for other engines to read the hudi table.
+
+#### 3.Lock Provider & Cocurrency Management
+
+[TBD]
+
+#### 4. Schema Evolution
+
+[TBD]
+
+#### 5. Cache
+
+Cache the frequently accessed data like table, timeline related information. For larger metadata, cache them with LRU strategy.
+
+The main points are:
+
+- what to cahe
+
+- how to cache
+
+- when to update cache data to keep consistency with the data in the storage.
+
+[TBD]
+
+#### 6. Authorization Management
+
+[TBD]
+
+## Implementation
```

Comment:
would love to see some plans on when/how to incorporate existing sync features like hive-sync, glue-sync, datahub-sync, bigquery/snowflake-sync. it would be a good idea to consolidate the sync features

Reply:
@minihippo cross-posting a related comment here: https://github.com/apache/hudi/pull/5064#issuecomment-1238907156 
clarifying the RFC plans will certainly help expedite the implementation's review process